applications/HDF5Application/custom_io/hdf5_container_component_io.cpp

```diff
@@ -386,6 +386,8 @@ ContainerComponentIO<TContainerType, TContainerItemType, TComponents...>::Contai
     for (std::size_t i = 0; i < num_components; ++i)
         mComponentNames[i] = Settings["list_of_variables"].GetArrayItem(i).GetString();
 
+    std::sort(mComponentNames.begin(), mComponentNames.end());
```

Comment:
May be itisa good idea to put a comment why the sorting is necessary.

Reply:
Yup, I just realized that too! + test case